### ISSUE TITLE
RadCalendar: Error thrown once I click on a date using viewMode="Year" and selectionMode="Range".

### ISSUE THREAD

**Issue by relvis** [telerik/nativescript-ui-feedback#494]
An error has been thrown when I click on a date using RadCalendar, the error reads: 

`file:///app/tns_modules/nativescript-pro-ui/calendar/calendar.js:1447:84: JS ERROR TypeError: null is not an object (evaluating 'this._owner.get().ios.selectedDatesRange.startDate')`

This is my configuration of the Calendar:

`<RadCalendar #calendar viewMode="Year" selectionMode="Range" [minDate]="minDate" [maxDate]="maxDate" (dateSelected)="onDateSelected($event)"></RadCalendar>`

Thanks!!!

![simulator screen shot - iphone x - 2018-01-18 at 15 18 29](https://user-images.githubusercontent.com/3420851/35119441-eed8054a-fc62-11e7-88ac-213b52f16b17.png)

**Comment by NickIliev** [comment]
@relvis I've tried to reproduce the issue using this example and introducing your settings (`selectionMode="Range" [minDate]="minDate" [maxDate]="maxDate" (dateSelected)="onDateSelected($event)"`) but everythings works as expected on my side.

Perhaps, you can create a demonstration that can reproduce the issue [in the NativeScript Playground](https://play.nativescript.org/) or provide a sample project via Github or archive.

**Comment by relvis** [comment]
Hello @NickIliev, I have shared a playground example with you, I hope it helps to track the issue...

[https://play.nativescript.org/?template=play-ng&id=KwLMRE](https://play.nativescript.org/?template=play-ng&id=KwLMRE)

Thanks!

**Comment by NickIliev** [comment]
@relvis thanks for providing the Playground demo - I can confirm that the above scenario is throwing on iOS only. Loggins this one as a bug - stay tuned for updates related to this issue.

Steps to reproduce: use [this Playground demo](https://play.nativescript.org/?template=play-ng&id=KwLMRE) on iOS device

Stack trace:
```
2018-02-01 13:54:14.734 nsplaydev[384:206277] JavaScript stack trace:
2018-02-01 13:54:14.736 nsplaydev[384:206277] 1 calendarDidSelectDate@file:///app/tns_modules/nativescript-pro-ui/calendar/calendar.js:1447:84
2018-02-01 13:54:14.736 nsplaydev[384:206277] 2 UIApplicationMain@[native code]
2018-02-01 13:54:14.736 nsplaydev[384:206277] 3 start@file:///app/tns_modules/tns-core-modules/application/application.js:235:26
2018-02-01 13:54:14.736 nsplaydev[384:206277] 4 bootstrapApp@file:///app/tns_modules/nativescript-angular/platform-common.js:86:28
2018-02-01 13:54:14.736 nsplaydev[384:206277] 5 bootstrapModule@file:///app/tns_modules/nativescript-angular/platform-common.js:72:26
2018-02-01 13:54:14.736 nsplaydev[384:206277] 6 anonymous@file:///app/main.js:8:57
2018-02-01 13:54:14.737 nsplaydev[384:206277] 7 evaluate@[native code]
2018-02-01 13:54:14.737 nsplaydev[384:206277] 8 moduleEvaluation@[native code]
2018-02-01 13:54:14.737 nsplaydev[384:206277] 9 @[native code]
2018-02-01 13:54:14.737 nsplaydev[384:206277] 10 promiseReactionJob@[native code]
2018-02-01 13:54:14.737 nsplaydev[384:206277] JavaScript error:
file:///app/tns_modules/nativescript-pro-ui/calendar/calendar.js:1447:84: JS ERROR TypeError: null is not an object (evaluating 'this._owner.get().ios.selectedDatesRange.startDate')
2018-02-01 13:54:14.739 nsplaydev[384:206277] PlayLiveSync: Uncaught Exception
2018-02-01 13:54:14.740 nsplaydev[384:206277] PlayLiveSync: Sending crash report
```

**Comment by relvis** [comment]
Hi guys, just checking out with you, I just installed `"nativescript-ui-calendar": "^3.5.0",`. The fix for this issue is coming in the next release? 
Thanks!

**Comment by zbranzov** [comment]
Released in v3.5.1.
